src/learn/PaywallModal.js

```diff
@@ -12,6 +13,25 @@ import Button from "react-bootstrap/lib/Button";
 
 function LoadingModal(props) {
   const [showPayment, setShowPayment] = useState(false);
+
+  const history = useHistory();
+  const modalRef = useRef();
+
+  useEffect(()=>{
+    let handleModalClose = (event) =>{
+      if(!modalRef.current?.contains(event.target)){
```

Comment:
This functionality is meant to say 'if you click outside of the area, then close modal go to home' - however, it didn't work for me. Can you try to re-test it? Reference the `Modal` component in the `components` folder, for a work example.

Reply:
It is working now!That 'showPayment' verification was breaking the feature.

![CPT2109041429-1152x664](https://user-images.githubusercontent.com/60116310/132103410-58644eee-9e4d-4980-83de-8fa3759e1a3b.gif)

Reply:
Is your code breaking when you click 'Purchase'? Mine is... on the production site.

Reply:
no, its working

![CPT2109041825-832x629](https://user-images.githubusercontent.com/60116310/132108433-cdb0f0f8-95f8-4306-9290-3fc6f30a9658.gif)

Reply:
That was a bug that I just fixed haha it was breaking in production for some reason.

Reply:
Variable propagation wasn't quite working, changed the stripeKey to a prop from paywall modal and it's all good